linera-sdk/wit/contract-runtime-api.wit

```diff
@@ -16,6 +16,9 @@ interface contract-runtime-api {
     create-application: func(module-id: module-id, parameters: list<u8>, argument: list<u8>, required-application-ids: list<application-id>) -> application-id;
     try-call-application: func(authenticated: bool, callee-id: application-id, argument: list<u8>) -> list<u8>;
     emit: func(name: stream-name, value: list<u8>) -> u32;
+    read-event: func(chain-id: chain-id, name: stream-name, index: u32) -> list<u8>;
+    subscribe-to-events: func(chain-id: chain-id, application-id: generic-application-id, name: stream-name);
+    unsubscribe-from-events: func(chain-id: chain-id, application-id: generic-application-id, name: stream-name);
```

Comment:
In the spirit of system calls, we could have a separate triplet of methods for each system stream (which currently is not even needed), then a single API for all user streams.

Reply:
In this way, `linera_execution::StreamId` could be like `Operation`:
```
enum StreamId {
    System(SystemStreamId),
    User(ApplicationId, Vec<u8>),
}
```

Reply:
I changed it to user app IDs in https://github.com/linera-io/linera-protocol/pull/3692/commits/04b1b74f52844ecd3c844e84e02d3d4c256d8bbf.

Reply:
(I didn't change the `StreamId` type yet, because I'll probably have to do that in a later PR anyway.)